modules/compactor/compactor.go

```diff
@@ -308,10 +309,13 @@ func countSpans(dataEncoding string, objs ...[]byte) int {
 
 		for _, b := range t.Batches {
 			for _, ilm := range b.ScopeSpans {
-				spans += len(ilm.Spans)
+				if len(ilm.Spans) > 0 {
+					level.Debug(log.Logger).Log("msg", "max size of trace exceeded", "traceId", tempoUtil.TraceIDToHexString(ilm.Spans[0].TraceId), "discarded_span_count", len(ilm.Spans))
```

Comment:
this will hit repeatedly for every batch/scope spans in the same trace. let's just move it to line 319 after the loop and only log once.

Reply:
yeap :) it's applied.